extensions/vertx-web/deployment/src/main/java/io/quarkus/vertx/web/deployment/devmode/VertxHotReplacementSetup.java

```diff
@@ -35,26 +36,39 @@ void handleHotReplacementRequest(RoutingContext routingContext) {
             routingContext.next();
             return;
         }
-        boolean restart = false;
-        synchronized (this) {
-            if (nextUpdate < System.currentTimeMillis() || hotReplacementContext.isTest()) {
+        routingContext.request().pause();
+        routingContext.vertx().executeBlocking(new Handler<Promise<Object>>() {
+            @Override
+            public void handle(Promise<Object> event) {
+                boolean restart = false;
+                synchronized (this) {
+                    if (nextUpdate < System.currentTimeMillis() || hotReplacementContext.isTest()) {
+                        try {
+                            restart = hotReplacementContext.doScan(true);
+                        } catch (Exception e) {
+                            throw new IllegalStateException("Unable to perform hot replacement scanning", e);
+                        }
+                        nextUpdate = System.currentTimeMillis() + HOT_REPLACEMENT_INTERVAL;
+                    }
+                }
+                if (hotReplacementContext.getDeploymentProblem() != null) {
+                    handleDeploymentProblem(routingContext, hotReplacementContext.getDeploymentProblem());
+                    return;
```

Comment:
you need to complete the promise.

Reply:
It's not necessarily required because you don't have any completion handler.

Reply:
ideally we should complete the promise and in the completion handler of execute blocking then resume the request

Reply:
Much better! Thanks @stuartwdouglas